### DISCUSSION TITLE
How to pass proxy configuration from sequelize to the snowflake connector

### DISCUSSION THREAD

**Discussion by TASimpson** [sequelize/sequelize#15601] [Help & Questions] [answered]
Hi! I have an open question on stackoverflow for this. But would love some guidance on how to set up the proxy configuration to send to Snowflake connector. This is my LAST missing piece and I cannot find what I need anywhere. 

I have the following setup based on what is described here: https://sequelize.org/docs/v6/other-topics/dialect-specific-things/#snowflake-experiment

```
SequelizeModule.forRoot({
      dialect: 'snowflake',
      host: 'localhost',
      dialectOptions: {
        account: 'account',
        application: 'SNOWFLAKE',
        role: 'role',
        warehouse: 'warehouse',
        schema: 'schema'
      },
      username: 'username',
      password: 'password',
      database: 'database',
      define : {
        freezeTableName: true
      },
      models: (...list of models here),
      quoteIdentifiers: false
    }), 
```

I have tried the following with NO success:

Added the following under dialectOptions

```
host: 'url to proxy'
port: 'port to proxy'
```

AND

```
stream : {
   host: 'url to proxy'
   port: 'port to proxy'
}
```

AND

```
proxyHost: 'url to proxy'
proxyPort: 'port to proxy'
```

PLEASE O PLEASE does anyone know how to set up the proxy in this setup to be passed to the snowflake connector proxy settings?? 

Thank you so much!

**Comment by WikiRik** [comment]
Did you already try passing the proxy information to the connector through the dialectOptions? We should pass them as is.

**Reply by TASimpson** [reply]
hey @WikiRik I am so sorry...I didnt copy EVERYTHING down. I updated what I have tried thus far with no luck. I may doing this completely wrong. But please let me know if one of the 3 options i tried should have worked. Or if I have my wording wrong. THANK YOU!

**Comment by ephys** [comment, marked as answer]
These are the options you can use to configure the proxy: https://docs.snowflake.com/en/user-guide/nodejs-driver-use.html#connecting-through-an-authenticated-proxy

As @WikiRik said, anything you add in `dialectOptions` will be sent as-is to `snowflake.createConnection`. Any option that `snowflake.createConnection` accepts can be used in `dialectOptions`

**Reply by TASimpson** [reply]
Thanks @ephys that was one of the options I passed (as seen in my 'what i tried') I will put this back and get with my team. I will let y'all know very soon if it didn't work again. If so, I guess that will become an issue to open. Thank you both for answering so quickly!

**Reply by WikiRik** [reply]
Feel free to let us know what comes out of the discussion with your team. If you think you need to open an issue you have to do it with the [snowflake connector](https://github.com/snowflakedb/snowflake-connector-nodejs) directly btw. Since we pass `dialectOptions` as-is to the `createConnection` function of the connector

**Reply by TASimpson** [reply]
It looks like it worked! I am not sure what happened when we did this the first time. But I updated our documentation to describe exactly what y'all told us! Thank you so much for a fast response and assistance!

**Reply by ephys** [reply]
Happy to hear that :)
